labs/docker-101/3.md

```diff
@@ -2,131 +2,126 @@
-The cloned repository contains two files:
+   * hellodevnet.py
+   * Dockerfile
 
-* hellodevnet.py
-* Dockerfile
+   The "hellodevnet.py" file is our custom application. It contains the following Python code:
 
-The "hellodevnet.py" file is our custom application. It contains the
-following Python code:
+     ``` python
+      #!/usr/bin/env python
+      print("Hello from DevNet!")
+     ```
 
-``` python
-#!/usr/bin/env python
+   The Dockerfile contains the instructions required to build the custom image:
 
-print("Hello from DevNet!")
-```
+    ```
+    FROM ubuntu
+    RUN apt-get update
+    RUN apt-get -y install python
+    COPY hellodevnet.py /hellodevnet.py
+    RUN ["chmod", "+x", "/hellodevnet.py"]
+    CMD ["/hellodevnet.py"]
+    ```
 
-The Dockerfile contains the instructions needed to build the custom
-image:
+   This Dockerfile says:
 
-```
-FROM ubuntu
-RUN apt-get update
-RUN apt-get -y install python
-ADD hellodevnet.py /hellodevnet.py
-CMD ["/hellodevnet.py"]
-```
+   1. `FROM ubuntu`  
+       Extend the existing ubuntu public docker image. Our previous examples built an image from scratch. In this case we begin with a previously-built container image and           extend with our own customisations.
 
-This Dockerfile says:
+   2. `RUN apt-get update`  
+       Ensure the package-management tools in the base Ubuntu container are updated to use the latest software.
 
-1. `FROM ubuntu`  
-   Extend the existing ubuntu public docker image. Our previous
-   examples built an image from scratch. In this case we begin with a
-   previously-built container image and extend with our own
-   customisations.
+   3. `RUN apt-get -y install python`  
+       Use apt-get to install Python and all its dependencies in the container. The reason to build this container on an existing Ubuntu image is that we could use apt-get to install needed software.
 
-2. `RUN apt-get update`  
-   Ensure the package-management tools in the base Ubuntu container
-   are updated to use the latest software.
+   4. `COPY hellodevnet.py /hellodevnet.py`
+       Copy the Python program from the local directory into the container as `/hellodevnet.py`.
+        
+   5. `RUN ["chmod", "+x", "/hellodevnet.py"]`
+       Grant permission to execute the `/hellodevnet.py` file
+      
+   6. `CMD ["/hellodevnet.py"]`  
+    Run the Python program when the container starts up.
+    
+    >**Note:** Instead of cloning this GitHub repository, you can also add the above mentioned python code in the hellodevnet.py file and Dockerfile commands in the Dockerfile using the `vi` command in the terminal and continue with step 2.
+   
+2. Build the docker image using the following command:
 
-3. `RUN apt-get -y install python`  
-   Use apt-get to install Python and all its dependencies in the
-   container. The reason to build this container on an existing Ubuntu
-   image was so that we could use apt-get to install needed software.
+    ```
+    docker build .
+    ```
+   
```

Comment:
Tell them to make note of the container ID that is returned once the build is done.